plugins/modules/consul_agent_check.py

```diff
@@ -0,0 +1,270 @@
+            If no suffix is supplied V(s) will be used by default, for example V(10) will be V(10s).
+          - Required if one of the parameters O(script), O(http), or O(tcp) is specified.
+    notes:
+        type: str
+        description:
+          - Notes to attach to check when registering it.
+    args:
+        type: str
+        description:
+          - Specifies command arguments to run to update the status of the check.
+          - Requires O(interval) to be provided.
+          - Mutually exclusive with O(ttl), O(tcp) and O(http).
+          - There is an issue with args. It throws an 'Invalid check: TTL must be > 0 for TTL checks'
+            https://github.com/hashicorp/consul/issues/6923#issuecomment-564476529
+    
+    ttl:
+        type: str
+        description:
+          - Checks can be registered with a TTL instead of a O(script) and O(interval)
+            this means that the service will check in with the agent before the
+            TTL expires. If it doesn't the check will be considered failed.
+            Required if registering a check and the script an interval are missing
+            Similar to the interval this is a number with a V(s) or V(m) suffix to
+            signify the units of seconds or minutes, for example V(15s) or V(1m).
+            If no suffix is supplied V(s) will be used by default, for example V(10) will be V(10s).
+          - Mutually exclusive with O(script), O(tcp) and O(http).
+    tcp:
+        type: str
+        description:
+          - Checks can be registered with a TCP port. This means that consul
+            will check if the connection attempt to that port is successful (that is, the port is currently accepting connections).
+            The format is V(host:port), for example V(localhost:80).
+          - Requires O(interval) to be provided.
+          - Mutually exclusive with O(script), O(ttl) and O(http).
+        version_added: '1.3.0'
+    http:
+        type: str
+        description:
+          - Checks can be registered with an HTTP endpoint. This means that consul
+            will check that the http endpoint returns a successful HTTP status.
+          - Requires O(interval) to be provided.
+          - Mutually exclusive with O(script), O(ttl) and O(tcp).
+    timeout:
+        type: str
+        description:
+          - A custom HTTP check timeout. The consul default is 10 seconds.
+            Similar to the interval this is a number with a V(s) or V(m) suffix to
+            signify the units of seconds or minutes, for example V(15s) or V(1m).
+            If no suffix is supplied V(s) will be used by default, for example V(10) will be V(10s).
+    token:
+        type: str
+        description:
+          - The token key identifying an ACL rule set. May be required to register services.
+    
+'''
+
+EXAMPLES = '''
+- name: register tcp check for service 'nginx'
```

Comment:
```suggestion
- name: Register tcp check for service 'nginx'
```
Task names should start with a capital letter.